packages/medusa-file-spaces/src/services/digital-ocean.js

```diff
@@ -84,16 +66,69 @@ class DigitalOceanService extends AbstractFileService {
     })
   }
 
+
   async getUploadStreamDescriptor(fileData) {
-    throw new Error("Method not implemented.")
+    this.updateAwsConfig()
+
+    const pass = new stream.PassThrough()
```

Comment:
question (mostly for my understanding): what is the use of a pass through stream here? Is it just to adhere to the interface (the return type)?

Reply:
I have used pass through here because we want to return a stream that we can write to while aws can also read from it. As far as i understand we need a duplex stream for that (writeable and readable stream). A pass through stream is just a duplex stream which passes whatever is written to it directly to the output. https://nodejs.org/api/stream.html#class-streampassthrough.

Reply:
sorry I just saw the notion doc and it makes a lot of sense 👍 . Thanks @pKorsholm !